src/plugins/intel_gna/src/gna_plugin.cpp

```diff
@@ -200,7 +200,31 @@ void GNAPlugin::copyInputData(T *dst,
     }
 }
 
-void GNAPlugin::ExportScores(void *ptr_dst,
+template <typename T>
+void GNAPlugin::UpscaleAndCast(T* dst_ptr, const uint8_t* input_ptr, const Precision& precision_in, const float& scale_factor) {
+    switch (precision_in) {
```

Comment:
Input precision is not changed during the loop. Don't need to check it on each iteration.

Reply:
Nadezhda is right.  Templating needs to be at a higher level of abstraction.